Added Swagger Documentation and API Versioning

LBHDTortora opened the pull request:
Merge all conflicts between master and swagger branch.
Kept logic and new implementation from master and added implementation of Swagger Doc Gen and API Versioning.

LBHmsaitz commented on .vscode/launch.json:
```diff
@@ -0,0 +1,47 @@
+{
```
This can be removed from the project

LBHDTortora replied:
Removed from the PR, will add on .gitignore

LBHmsaitz commented on .vscode/tasks.json:
```diff
@@ -0,0 +1,15 @@
+{
```
This can be removed as well, it also could be added to the gitignore file

LBHDTortora replied:
Removed from the PR, will add on .gitignore

LBHmsaitz commented on property-api/Startup.cs:
```diff
@@ -22,11 +32,75 @@ public Startup(IConfiguration configuration)
+
+            services.AddSingleton<IApiVersionDescriptionProvider, DefaultApiVersionDescriptionProvider>();
+
+            services.AddApiVersioning(o =>
```
Suggest moving the swagger config to private methods to make the ConfigureServices method easier to read

LBHDTortora replied:
Removed from the PR, will add on .gitignore

LBHDTortora replied:
Method has been extracted. There was a conflict with the HttpGet method in Swagger with an explicit route triggering an exception. I have removed the route from the HttpGet attribute and kept the route on the [Route] attribute. This seems to have fixed the problem.

LBHDTortora commented:
Implemented Swagger Doc and Gen to Property API and included xml documentation to be generated on controllers methods.
Fixed a swagger conflicts with Http method where the [HttpGet] attribute could not include routing params (ex. [HttpGet('{params}')].
Params are still declared in the Route attribute. [Route('{params}')].
